### ISSUE TITLE
I'm lost understanding the command syntax. Any examples of what NAME is and what VALUE is?

### ISSUE THREAD

**Issue by yerrag** [Magisk-Modules-Repo/MagiskHidePropsConf#159]
I assume NAME refers to the device name such as SM-G920F for item 337 in the Certified Fingerprints List.
And I assume VALUE refers to the whole set of characters after the "=" sign in item 337

Is that a correct assumption?

So when using terminal with su, I type 

props SM-920F samsung/zerofltexx/zeroflte:7.0/NRD90M/G920FXXS5EQL1:user/release-keys

but I don't get a confirmation of success, but rather many lines of characters that don't make sense.

I know though that it didn't go thru because my fingerprint did not change. 

I know I'm doing something wrong but the documentation available hasn't helped me as I did try making sense of what the syntax means.

Any clarification would help. I appreciate it.

**Comment by MattWaller** [comment]
What I've done is this:
<br />
<code>
$ adb shell su -c "find -name \"Samsung.sh\""
</code>
<code>
find: ./storage/emulated: Permission denied
</code>
<code>
./dev/g3WtQr/.magisk/modules/MagiskHidePropsConf/printfiles/Samsung.sh
</code>
<code>
find: './dev/g3WtQr/.magisk/mirror/system_root': loop detected
</code>
<code>
./dev/g3WtQr/.magisk/mirror/data/adb/modules/MagiskHidePropsConf/printfiles/Samsung.sh
^C
</code>
<br />
This will give you the folder of where magisk is located, along with its modules. Open up Samsung.sh / copy it and it will give you the formatting you're looking for. "/dev/g3WtQr/.magisk/modules/MagiskHidePropsConf/printfiles/Samsung.sh"
<br />
**Note: the folder name will always change on reboot of the device so its not a static folder.** 
Just cp the .sh file to /sdcard/Download/Samsung.sh & pull it and you'll have the sh file. Then once you want to add your custom fingerprint you can append it to this sh or create another sh & move it back into the path where printfiles exists and when you run props you will see it as an option.
<br />
<code>
Samsung Galaxy A01 Core (10):Samsung:SM-A013G=samsung/a01corextc/a01core:10/QP1A.190711.020/A013GXXU1ATGE:user/release-keys__2020-07-01
</code>
<br />
<code>
Samsung Galaxy A3 2015 SM-A300FU (6.0.1):Samsung:SM-A300FU=samsung/a3ultexx/a3ulte:6.0.1/MMB29M/A300FUXXU1CRF1:user/release-keys
</code>
<br />
<code>
Samsung Galaxy A3 2016 SM-A310F (7.0):samsung:SM-A310F=samsung/a3xeltexx/a3xelte:7.0/NRD90M/A310FXXU4CSB2:user/release-keys__2019-02-01
</code>
<br />
<code>
Samsung Galaxy A3 2017 SM-A320FL (8.0.0):samsung:SM-A320FL=samsung/a3y17ltexc/a3y17lte:8.0.0/R16NW/A320FLXXU4CSD2:user/release-keys__2019-04-01
</code>
<br />
<code>
Samsung Galaxy A5 2015 SM-A500FU (6.0.1):Samsung:SM-A500FU=samsung/a5ultexx/a5ulte:6.0.1/MMB29M/A500FUXXU1CPH3:user/release-keys
</code>
<br />
<code>
Samsung Galaxy A5 2016 SM-A510F (7.0):samsung:SM-A510F=samsung/a5xeltexx/a5xelte:7.0/NRD90M/A510FXXS8CSF3:user/release-keys__2019-07-01
</code>
<br />
<code>
Samsung Galaxy A5 2017 SM-A520F (8.0.0):Samsung:SM-A520F=samsung/a5y17ltexx/a5y17lte:8.0.0/R16NW/A520FXXUGCTI9:user/release-keys__2020-09-01
</code>
<br />
<code>
Samsung Galaxy A6 SM-A600F (10):Samsung:SM-A600F=samsung/a6ltexx/a6lte:10/QP1A.190711.020/A600FNXXS7CTI1:user/release-keys__2020-09-01
</code>
<br />
<code>
Samsung Galaxy A6 Plus SM-A605G (9):Samsung:SM-A605G=samsung/a6pltedx/a6plte:9/PPR1.180610.011/A605GDXU3BSC6:user/release-keys__2019-03-01
</code>
<br />
<code>
Samsung Galaxy A7 2017 SM-A720F (8.0.0):Samsung:SM-A720F=samsung/a7y17ltexx/a7y17lte:8.0.0/R16NW/A720FXXU9CSK3:user/release-keys__2019-12-01
</code>
<br />
<code>
Samsung Galaxy A7 2018 SM-A750GN (9 & 10):samsung:SM-A750GN=samsung/a7y18ltedx/a7y18lte:9/PPR1.180610.011/A750GNDXU2BSD2:user/release-keys__2019-04-01;samsung/a7y18ltedx/a7y18lte:10/QP1A.190711.020/A750GNDXU6CUA1:user/release-keys__2021-01-01
</code>
<br />
<code>
Samsung Galaxy A8 Plus SM-A730F (7.1.1):Samsung:SM-A730F=samsung/jackpot2ltexx/jackpot2lte:7.1.1/NMF26X/A730FXXU2ARD1:user/release-keys
</code>
<br />
<code>
Samsung Galaxy A20 SM-A205W (9):Samsung:SM-A205W=samsung/a20cs/a20:9/PPR1.180610.011/A205WVLU3ASK2:user/release-keys__2019-11-01
</code>
<br />
<code>
Samsung Galaxy A40 SM-A405FN (10):Samsung:SM-A405FN=samsung/a40xx/a40:10/QP1A.190711.020/A405FNXXS3BTI3:user/release-keys__2020-09-01
</code>
<br />
<code>
Samsung Galaxy A50 SM-A505F (9):Samsung:SM-A505F=samsung/a50dd/a50:9/PPR1.180610.011/A505FDDU1ASD6:user/release-keys__2019-04-01
</code>
<br />
<code>
Samsung Galaxy A50S SM-A507FN (10):Samsung:SM-A507FN=samsung/a50sxx/a50s:10/QP1A.190711.020/A507FNXXS5BTI1:user/release-keys__2020-09-01
</code>
<br />
<code>
Samsung Galaxy A51 SM-A515F (10):Samsung:SM-A515F=samsung/a51nsxx/a51:10/QP1A.190711.020/A515FXXU1ASKJ:user/release-keys__2019-12-01
</code>
<br />
These are some samples of some fingerprints that are in Samsung.sh.

**Comment by MattWaller** [comment]
actually theres an easier command.
</br>
<code>
adb shell props SM-G986B samsung/y2sxeea/y2s:10/QP1A.190711.020/G986BXXU1ATCT:user/release-keys__2020-04-01
</code>
</br>
formatting should be:
</br>
props "Name" "fingerprint"

**Comment by yerrag** [comment]
Hi Matt, 

Thanks a lot! It looks like I was in the right direction. I had the right idea in the first place. The command wasn't wrong, but I was getting messages that seemed to be asking me for an answer, without which the command wouldn't be implemented, and since I didn't know how to respond, the fingerprint change wasn't carried out.

I have to either decrypt the messages and know how to answer so I don't get an error message, or I would have to customize a fingerprint myself. I should go back to your first message to figure out how to modify the fingerprint.

What I wanted to do was to change my actual fingerprint from that of a Samsung Galaxy S5 to that of a Note 4, as I wanted to fool an app into recognizing the putatively hidden spO2 sensor in an S5, which is recognized in a Note 4. I have heard through the rumor mill that the S5 and the Note 4 shares the same heartrate/spO2 sensor hardware, the difference being that the S5's spO2 sensor is merely software-disabled.

**Comment by yerrag** [comment]
This is the message I get when I try changing the fingerprint using the prop NAME VALUE command :
#[i] props sm-g920f samsung/zerofltex...

Loading... Please wait.

[H[J[H[J
[01;37mMagiskHide Props Config v6.1.2[00;37;40m
[01;37mby Didgeridoohan @ XDA Developers[00;37;40m

[01;33m=====================================[00;37;40m
 [01;36msm-g920f[00;37;40m
[01;33m=====================================[00;37;40m

This will set [01;36msm-g920f[00;37;40m to:

[01;36msamsung/zerofltexx/zeroflte:7.0/NRD90M/G920FXXS5EQL1:user/release-keys[00;37;40m

The currently set value is:
[01;36mCurrently not set, or the value could not be retrieved.[00;37;40m
Please enter the value to set.

Pick an option below to change
what boot stage the prop will
be set in, or set/reset a delay:

[01;32m1[00;37;40m - Default [01;32m(current)[00;37;40m
[01;32m2[00;37;40m - post-fs-data
[01;32m3[00;37;40m - late_start service
[01;32m4[00;37;40m - Both boot stages
[01;32md[00;37;40m - Delay

Do you want to continue?

Enter [01;32my[00;37;40m(es) or [01;32mn[00;37;40m(o)
or an option from above: 
[01;31mInvalid input[00;37;40m. Only enter '[01;32my[00;37;40m' or '[01;32mn[00;37;40m'.
[H[J
[01;37mMagiskHide Props Config v6.1.2[00;37;40m
[01;37mby Didgeridoohan @ XDA Developers[00;37;40m

[01;33m=====================================[00;37;40m
 [01;36msm-g920f[00;37;40m
[01;33m=====================================[00;37;40m

This will set [01;36msm-g920f[00;37;40m to:

[01;36msamsung/zerofltexx/zeroflte:7.0/NRD90M/G920FXXS5EQL1:user/release-keys[00;37;40m

The currently set value is:
[01;36mCurrently not set, or the value could not be retrieved.[00;37;40m
Please enter the value to set.

Pick an option below to change
what boot stage the prop will
be set in, or set/reset a delay:

[01;32m1[00;37;40m - Default [01;32m(current)[00;37;40m
[01;32m2[00;37;40m - post-fs-data
[01;32m3[00;37;40m - late_start service
[01;32m4[00;37;40m - Both boot stages
[01;32md[00;37;40m - Delay

Do you want to continue?

Enter [01;32my[00;37;40m(es) or [01;32mn[00;37;40m(o)
or an option from above: 
[01;31mInvalid input[00;37;40m. Only enter '[01;32my[00;37;40m' or '[01;32mn[00;37;40m'.
[H[J
[01;37mMagiskHide Props Config v6.1.2[00;37;40m
[01;37mby Didgeridoohan @ XDA Developers[00;37;40m

[01;33m=====================================[00;37;40m
 [01;36msm-g920f[00;37;40m
[01;33m=====================================[00;37;40m

This will set [01;36msm-g920f[00;37;40m to:

[01;36msamsung/zerofltexx/zeroflte:7.0/NRD90M/G920FXXS5EQL1:user/release-keys[00;37;40m

The currently set value is:
[01;36mCurrently not set, or the value could not be retrieved.[00;37;40m
Please enter the value to set.

Pick an option below to change
what boot stage the prop will
be set in, or set/reset a delay:

[01;32m1[00;37;40m - Default [01;32m(current)[00;37;40m
[01;32m2[00;37;40m - post-fs-data
[01;32m3[00;37;40m - late_start service
[01;32m4[00;37;40m - Both boot stages
[01;32md[00;37;40m - Delay

Do you want to continue?

Enter [01;32my[00;37;40m(es) or [01;32mn[00;37;40m(o)
or an option from above: 
[01;31mInvalid input[00;37;40m. Only enter '[01;32my[00;37;40m' or '[01;32mn[00;37;40m'.
[H[J
[01;37mMagiskHide Props Config v6.1.2[00;37;40m
[01;37mby Didgeridoohan @ XDA Developers[00;37;40m

[01;33m=====================================[00;37;40m
 [01;36msm-g920f[00;37;40m
[01;33m=====================================[00;37;40m

This will set [01;36msm-g920f[00;37;40m to:

[01;36msamsung/zerofltexx/zeroflte:7.0/NRD90M/G920FXXS5EQL1:user/release-keys[00;37;40m

The currently set value is:
[01;36mCurrently not set, or the value could not be retrieved.[00;37;40m
Please enter the value to set.

Pick an option below to change
what boot stage the prop will
be set in, or set/reset a delay:

[01;32m1[00;37;40m - Default [01;32m(current)[00;37;40m
[01;32m2[00;37;40m - post-fs-data
[01;32m3[00;37;40m - late_start service
[01;32m4[00;37;40m - Both boot stages
[01;32md[00;37;40m - Delay

Do you want to continue?

it keeps repeating the same message .

**Comment by yerrag** [comment]
From what I am to gather, I am being asked to answer from 5 options, and if it wasn't getting an answer, it is asking me if I wanted to continue or not. So far I have tried answering with typing "1" and "y" each time, and each time I get an "I/O error" so command is aborted.

**Comment by yerrag** [comment]
These are two commands I have typed:

To emulate a Galaxy S6: props SM-920F samsung/zerofltexx/zeroflte:7.0/NRD90M/G920FXXS5EQL1:user/release-keys

To emulate a Note 4: props SM-N910F samsung/trltexx/trlte:6.0.1/MMB29M/N910FXXS1DQH9:user/release-keys

Both failing to successfully change the fingerprint of my Samsung SM-G900T

**Comment by MattWaller** [comment]
I had better success modifying the Samsung.sh file and just loading in the singular profile. 
This is what i coded in python.
[finerprintsetter.py](https://github.com/MattWaller/fingerprintsetter/blob/main/fingerprintsetter.py)

I had a file labeled good_records.txt where I have my user agents & it randomly selects one. Running this script will automatically reset the phone with the desired fingerprint. It does delete your existing Samsung.sh fingerprints, but if you're looking to automate it / retool it go for it.

**Comment by yerrag** [comment]
I'm gonna try modifying the samsung.sh file manually. I don't know how to program much but I get what you're saying. I'll do it the manual way. I hope using the terminal with su will do it. Appreciate your help Matt!!

**Comment by yerrag** [comment]
I came to realize that I was using QUTE terminal emulator and that the gobbledygook I was getting as posted 5 post earlier was coming from the way QUTE was parsing the messages.

I used a terminal that came along with a rom package I installed (a resurrection pie build) on my s906s.

I used the same command posted 3 posts earlier to emulate an SM-910f, and finally, I could understand the characters coming out, and the process went smoothly.

I apologize for assuming any terminal emulator would just do. But thanks Matt for giving me another way out in case this didn't work. All along the software has been fine. It's just me being a newbie all along!
